### DISCUSSION TITLE
New PR with a few cool features, please chime in and let me know what you think

### DISCUSSION THREAD

**Discussion by altryne** [hlky/stable-diffusion-webui#422] [Show and tell]
(Upcoming features in #421 )

I've rewritten this feature like 3 times but I'm so glad I added it. 
## I give you: dynamic params 

You can now control the basics params right from the input box, and they will update in the system : 

![CleanShot 2022-09-01 at 00 43 13](https://user-images.githubusercontent.com/463317/187849190-433d92f8-b103-4b18-907b-d287a114833a.gif)

This also means, we can copy paste and load most of the params from both the old SD discord bot replies and the new gobot replies 

This also means, users are free to specify whichever params they want outside of our defaults. 

Supports many way users can think about params 
`width:124` or `-W 234` or `--cfg-scale 123` etc etc 

## Revamped output box, with simplified UI and colorful tags for params : 
![CleanShot 2022-09-01 at 00 46 26@2x](https://user-images.githubusercontent.com/463317/187849618-ddcf5183-638c-4903-9990-6455444a0ae9.jpg)

## Mobile responsive support and prompt input changes: (not even dreamstudio has this!) 
* The input  now switches to multiline when the prompt is too big, to support the longer prompts. 
* Mobile now looks much better, the results are close to the prompt and the propt is multiline 
* Added a subsection in css for mobile fixes and fixed button rows 
* 
![CleanShot 2022-09-01 at 00 47 50@2x](https://user-images.githubusercontent.com/463317/187850033-dcdbc2b3-5a44-419a-9881-b716b27b0ff6.jpg)

## Quality of life changes  and sensible defaults 
* Hid the batch_* thing inside advanced settings (as this screws most users up and confuses them) 
* renamed the batch_size to number (like in dreamstudio) and it's scroll size is much nicer now
* Lowered the basic width / height to 1024 (let's admit, most folks can't generate higher than that) 
* Updated the playground to generate rando images with supporting the provided width and height 

TODOS - post merge: 
Restore the aspect ratio calculation 
Restore the loader css missing. 
Sleep

**Comment by athu16** [comment]
This looks great! But I didn't get a couple of things
> Hid the batch_* thing inside advanced settings (as this screws most users up and confuses them)

Do you mean both batch_size and batch_count? I feel batch_count could remain on the main screen
Also I think instead of renaming batch_size to "number", you could do it with batch_count, as people who don't know batch_size significantly increases VRAM usage will get confused after running into OOM errors

**Reply by codedealer** [reply]
This! I've been staring at the webui constantly for the last few days and I still have to take a second sometimes to discern between "batch_count" and "batch_size".

* Rename "batch_count" to count and leave it in simple settings.
* Move memory hungry "bath_size" to advanced ones.

Also as a note, the names of some buttons are overly verbose, I wonder if we should give them noob friendly simpler names and hide the description of each feature in tooltips that show when you hover (tap ❔ on mobile)?

**Reply by altryne** [reply]
Felt much cleaner once I removed it to advanced settings and renamed it to align with dream studio

**Comment by sedetweiler** [comment]
Great work!  It would be pretty amazing if you could also add the prompt and other settings to the image data.  IPTC has a lot of fields in there that would work well for this and we could then have shorter image file names.

**Reply by altryne** [reply]
Feels like this odd outside of the scope but definitely
